overlord/aspectstate/aspectstate_test.go

```diff
@@ -196,3 +197,146 @@ func (s *aspectTestSuite) TestNewTransactionNoState(c *C) {
 		c.Assert(value, Equals, "bar")
 	}
 }
+
+type filterSampleSuite struct {
+	state *state.State
+	bag   aspects.JSONDataBag
+}
+
+var _ = Suite(&filterSampleSuite{})
+
+func (s *filterSampleSuite) SetUpTest(c *C) {
+	var raw map[string]json.RawMessage
+	err := json.Unmarshal([]byte(`{
+		"snaps": {
+			"test-snapd-sh": {
+				"name":   "test-snapd-sh",
+				"status": "installed"
+			},
+			"core20": {
+				"name":   "core20",
+				"status": "active"
+			},
+			"snapcraft": {
+				"name":   "snapcraft",
+				"status": "active"
+			},
+			"firefox": {
+				"name":   "firefox",
+				"status": "active"
+			},
+			"snapd": {
+				"name":   "snapd",
+				"status": "active"
+			},
+			"vlc": {
+				"name":   "vlc",
+				"status": "inactive"
+			},
+			"spotify": {
+				"name":   "spotify",
+				"status": "failed"
+			},
+			"discord": {
+				"name":   "discord",
+				"status": "active"
+			},
+			"shellcheck": {
+				"name":   "shellcheck",
+				"status": "active"
+			},
+			"htop": {
+				"name":   "htop",
+				"status": "inactive"
+			}
+		}
+	}`), &raw)
+	c.Assert(err, IsNil)
+
+	s.state = overlord.Mock().State()
+	s.bag = aspects.JSONDataBag(raw)
+}
+
+func (s *filterSampleSuite) TestQueryNoFilters(c *C) {
+	var res interface{}
+	err := aspectstate.QueryAspect(s.bag, "acc", "bundle", "asp", "snaps", "", &res)
+	c.Assert(err, IsNil)
+	// returns all snaps
+	obj, ok := res.(map[string]json.RawMessage)
+	c.Assert(ok, Equals, true)
+	c.Assert(obj, HasLen, 10)
+
+	discordRaw, ok := obj["discord"]
+	c.Assert(ok, Equals, true)
+
+	var discord map[string]json.RawMessage
+	err = json.Unmarshal(discordRaw, &discord)
+	c.Assert(err, IsNil)
+
+	assertField(c, discord, "name", "discord")
+	assertField(c, discord, "status", "active")
+
+	htopRaw, ok := obj["htop"]
+	c.Assert(ok, Equals, true)
+
+	var htop map[string]json.RawMessage
+	err = json.Unmarshal(htopRaw, &htop)
+	c.Assert(err, IsNil)
+
+	assertField(c, htop, "name", "htop")
+	assertField(c, htop, "status", "inactive")
+}
+
+func (s *filterSampleSuite) TestQueryFilterNameWithParameter(c *C) {
+	var res interface{}
+	err := aspectstate.QueryAspect(s.bag, "acc", "bundle", "asp", "snaps", "name=firefox", &res)
+	c.Assert(err, IsNil)
+	obj, ok := res.(map[string]json.RawMessage)
+	c.Assert(ok, Equals, true)
+	c.Assert(obj, HasLen, 2)
+	assertField(c, obj, "name", "firefox")
+	assertField(c, obj, "status", "active")
+}
+
+func (s *filterSampleSuite) TestQueryFilterNameWithRequest(c *C) {
+	var res interface{}
+	err := aspectstate.QueryAspect(s.bag, "acc", "bundle", "asp", "snaps.firefox", "", &res)
+	c.Assert(err, IsNil)
+
+	obj, ok := res.(map[string]json.RawMessage)
+	c.Assert(ok, Equals, true)
+	c.Assert(obj, HasLen, 2)
+	assertField(c, obj, "name", "firefox")
+	assertField(c, obj, "status", "active")
+}
+
+func (s *filterSampleSuite) TestQueryFilterStatus(c *C) {
+	var res interface{}
```

Comment:
can res here be:

var res map[string]map[string]interface{}

directly? or that doesn't work atm